src/core/probe/kernel/mod.rs

```diff
@@ -0,0 +1,26 @@
+//! # Kernel probes
+//!
+//! Module providing an API to attach probes in the Linux kernel, e.g. using
+//! kprobes and raw tracepoints. The need to attach a probe in the kernel can
+//! come from various sources (different collectors, the user, etc) and as such
+//! some kind of synchronization and common logic is required; which is provided
+//! here.
+
+#![allow(dead_code)] // FIXME
+
+use anyhow::Result;
+
+/// Main object representing the kernel probes and providing an API for
+/// consumers to register probes, hooks, maps, etc.
+pub(crate) struct Kernel {}
+
+impl Kernel {
```

Comment:
General question. What should be our policy with regards to `mod.rs`?

I see that [recent changes to rust-lang have made the use of `mod.rs` not needed](https://doc.rust-lang.org/beta/edition-guide/rust-2018/path-changes.html#no-more-modrs).

Particularly, I find it annoying to have actual code (not just relatively specific things like re-exports and includes) in a file that's named the same as many others. I end up having a bunch of `mod.rs` files open in my editor and it's confusing. It feels to me like putting important code into `__init__.py`. Maybe it's just me and I just have to get used to it, although it does seem like[ one (small) reason behind the 2018 shift](https://github.com/rust-lang/rfcs/blob/master/text/2126-path-clarity.md#the-modrs-file), so I'm not alone, phew :-)

What do you guys think?
/cc @vlrpl

Some additional reading:
https://blog.devgenius.io/how-to-mod-in-rust-b2da3e1ff0f2

A nice image:
https://i.redd.it/1yy98srxyvx81.png

Reply:
So, are you planning on updating the PR or should we switch to mod.rs re-exporting after?